common/buildcraft/energy/generation/OilGenerator.java

```diff
@@ -104,17 +104,21 @@ public void generate(Random random, int chunkX, int chunkZ, World world, IChunkG
         boolean oilBiome = BCEnergyConfig.surfaceDepositBiomes.contains(biome.getRegistryName());
 
         double bonus = oilBiome ? 3.0 : 1.0;
+        bonus *= BCEnergyConfig.oilWellGenerationRate;
         if (BCEnergyConfig.excessiveBiomes.contains(biome.getRegistryName())) {
             bonus *= 30.0;
         }
         final GenType type;
-        if (rand.nextDouble() <= CHANCE_LARGE * bonus) {
+
+        final double random_number = rand.nextDouble();
+
+        if (random_number <= CHANCE_LARGE * bonus) {
             // 0.04%
             type = GenType.LARGE;
-        } else if (rand.nextDouble() <= CHANCE_MEDIUM * bonus) {
+        } else if (random_number <= CHANCE_MEDIUM * bonus) {
             // 0.1%
             type = GenType.MEDIUM;
-        } else if (oilBiome && rand.nextDouble() <= CHANCE_LAKE * bonus) {
+        } else if (oilBiome && random_number <= CHANCE_LAKE * bonus) {
```

Comment:
I wanted to change this originally, however this approach has a few problems:
 - We break existing worldgen. This will only be noticeable if an oil well goes across a chunk boundary, or if someone compares two oil spouts in different worlds. Obviously this was originally broken from 1.7.10 when I added the `MAGIC_GEN_NUMBER`, but I think it's more of an issue for newer worlds.
 - The chance constants (`CHANCE_LAKE`, `CHANCE_MEDIUM`, `CHANCE_LARGE`) need to be changed so that they also include the previously tested value. Something like this:
````java
     private static final double CHANCE_LARGE = 0.0004;
     private static final double CHANCE_MEDIUM = 0.001 + CHANCE_MEDIUM;
     private static final double CHANCE_LAKE = 0.02 + CHANCE_LAKE;
````
If/when these constants are available in the config it would probably be a good idea to do this addition automatically, and complain if the final number is greater than 1.

However I don't think the (tiny) performance gain is much of a gain compared to world/chunk breaks.

Reply:
Very good catch. I did not think of the update on the RNG state on each call (besides the bad math). Will revert it.

Reply:
Yeah it's an annoying case of something that begs to be fixed but results in breaking compat if we do :/

Reply:
Sorry for constantly changing my mind on this, but it's probably safer to leave it as-is (not adding a variable) and leaving it to be cleaned up later, probably alongside the additional config options.